resolver.go

```diff
@@ -314,3 +327,30 @@ func (r *Resolver) DialContext(ctx context.Context, network, addr string) (net.C
 	}
 	return nil, lastErr
 }
+
+func (r *Resolver) AddZone(name string, zone Zone) error {
+	r.zonesMutex.Lock()
+	defer r.zonesMutex.Unlock()
+
+	if _, ok := r.Zones[name]; ok {
+		return fmt.Errorf(ErrExistingZoneFmt, name)
+	}
+
+	r.Zones[name] = zone
+	return nil
+}
+
+func (r *Resolver) GetZone(name string) (Zone, bool) {
+	r.zonesMutex.RLock()
+	defer r.zonesMutex.RUnlock()
+
+	zone, ok := r.Zones[name]
+	return zone, ok
```

Comment:
```suggestion
	return r.Zones[name]
```

Reply:
I don't believe map access supports returning the multiple assignment form and since the first return argument is not a pointer, we need to return both the value and the boolean.

Reply:
Got it. Makes sense!